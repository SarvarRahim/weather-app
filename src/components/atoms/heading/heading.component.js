/* eslint-disable react/jsx-pascal-case */
import React from 'react';

import { StyledHeader } from './heading.style';

export const Heading = ({ variant, children, color }) => {
  const HeadingComponent = StyledHeader[variant] || StyledHeader.h1;
  return (
    <HeadingComponent
      aria-level={variant?.slice(-1) || '1'}
      role='heading'
      color={color}
    >
      {children}
    </HeadingComponent>
  );
};
