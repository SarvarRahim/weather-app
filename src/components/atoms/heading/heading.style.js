import styled from 'styled-components';

const StyledHeader = {};

const H1 = styled.h1`
  font-size: 2.6rem;
  line-height: 3.5rem;
  font-family: 'ManropeSemiBold';
  color: ${props => props.color};
`;

const H2 = styled.h2`
  font-size: 2.4rem;
  line-height: 3.2rem;
  font-family: 'ManropeSemiBold';
  color: ${props => props.color};
`;

const H3 = styled.h3`
  font-size: 2rem;
  line-height: 2.4rem;
  font-family: 'ManropeMedium';
  color: ${props => props.color};
`;

const H4 = styled.h4`
  font-size: 1.8rem;
  line-height: 2.2rem;
  font-family: 'ManropeSemiBold';
  color: ${props => props.color};
`;

const H5 = styled.h5`
  font-size: 1.6rem;
  line-height: 2rem;
  font-family: 'ManropeSemiBold';
  color: ${props => props.color};
`;

const H6 = styled.h6`
  font-size: 1.4rem;
  line-height: 1.8rem;
  font-family: 'ManropeMedium';
  color: ${props => props.color};
`;

StyledHeader.h1 = H1;
StyledHeader.h2 = H2;
StyledHeader.h3 = H3;
StyledHeader.h4 = H4;
StyledHeader.h5 = H5;
StyledHeader.h6 = H6;

export { StyledHeader };
