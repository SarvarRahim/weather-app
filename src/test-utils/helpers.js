import { render } from '@testing-library/react';
import React, { useRef } from 'react';
import { BrowserRouter, Router } from 'react-router-dom';
import { Provider as ReduxProvider } from 'react-redux';
import { ThemeProvider } from 'styled-components';
import { theme } from '@styles/Theme';
import { store } from '@store';
import { Formik } from 'formik';
import { VideoContext } from '@courses/routes/courses-single/course.context';
import { createMemoryHistory } from 'history';

const AllTheProviders = ({ children, formikValues = {} }) => {
  const videoRef = useRef();
  return (
    <BrowserRouter>
      <ReduxProvider store={store}>
        <ThemeProvider theme={theme}>
          <Formik initialValues={formikValues}>
            <VideoContext.Provider value={videoRef}>
              {children}
            </VideoContext.Provider>
          </Formik>
        </ThemeProvider>
      </ReduxProvider>
    </BrowserRouter>
  );
};

export const customRender = (ui, options) =>
  render(ui, { wrapper: AllTheProviders, ...options });

export const renderWithRouter = (
  ui,
  {
    route = '/',
    history = createMemoryHistory({ initialEntries: [route] })
  } = {}
) => {
  return {
    ...render(<Router history={history}>{ui}</Router>),
    history
  };
};
