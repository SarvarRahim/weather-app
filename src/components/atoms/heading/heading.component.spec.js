import { customRender } from '@test-utils/helpers';
import { screen } from '@testing-library/react';
import 'jest-styled-components';

import { Heading } from './heading.component';

const setup = ({ variant, children, color }) => {
  return customRender(
    <Heading variant={variant} color={color}>
      {children}
    </Heading>
  );
};

test('Render Heading1', () => {
  const { asFragment } = setup({ variant: 'h1' });
  expect(asFragment).toMatchSnapshot();
});

test('Render Heading2', () => {
  const { asFragment } = setup({ variant: 'h2' });
  expect(asFragment).toMatchSnapshot();
});

test('Render Heading3', () => {
  const { asFragment } = setup({ variant: 'h3' });
  expect(asFragment).toMatchSnapshot();
});

test('Render Heading4', () => {
  const { asFragment } = setup({ variant: 'h4' });
  expect(asFragment).toMatchSnapshot();
});

test('Render Heading5', () => {
  const { asFragment } = setup({ variant: 'h5' });
  expect(asFragment).toMatchSnapshot();
});

test('Render Heading6', () => {
  const { asFragment } = setup({ variant: 'h6' });
  expect(asFragment).toMatchSnapshot();
});

/// test props

test('Should get props for heading 1', () => {
  setup({ variant: 'h1', color: 'black', children: 'Title' });
  const headingElement = screen.getByRole('heading', { level: 1 });
  expect(headingElement).toBeTruthy();
});

test('Should get props for heading 2', () => {
  setup({ variant: 'h2', color: 'black', children: 'Title' });
  const headingElement = screen.getByRole('heading', { level: 2 });
  expect(headingElement).toBeTruthy();
});

test('Should get props for heading 3', () => {
  setup({ variant: 'h3', color: 'black', children: 'Title' });
  const headingElement = screen.getByRole('heading', { level: 3 });
  expect(headingElement).toBeTruthy();
});

test('Should get props for heading 4', () => {
  setup({ variant: 'h4', color: 'black', children: 'Title' });
  const headingElement = screen.getByRole('heading', { level: 4 });
  expect(headingElement).toBeTruthy();
});

test('Should get props for heading 5', () => {
  setup({ variant: 'h5', color: 'black', children: 'Title' });
  const headingElement = screen.getByRole('heading', { level: 5 });
  expect(headingElement).toBeTruthy();
});

test('Should get props for heading 6', () => {
  setup({ variant: 'h6', color: 'black', children: 'Title' });
  const headingElement = screen.getByRole('heading', { level: 6 });
  expect(headingElement).toBeTruthy();
});
