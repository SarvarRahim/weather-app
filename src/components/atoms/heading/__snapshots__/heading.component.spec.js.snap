// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Heading1 1`] = `[Function]`;

exports[`Render Heading2 1`] = `[Function]`;

exports[`Render Heading3 1`] = `[Function]`;

exports[`Render Heading4 1`] = `[Function]`;

exports[`Render Heading5 1`] = `[Function]`;

exports[`Render Heading6 1`] = `[Function]`;
